usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAgencyLangAndFeedInfoFeedLangMatch.java

```diff
@@ -0,0 +1,95 @@
+ */
+public class ValidateAgencyLangAndFeedInfoFeedLangMatch {
+    private static final String MUL = "mul";
+    private final GtfsDataRepository dataRepo;
+    private final ValidationResultRepository resultRepo;
+    private final Logger logger;
+
+    /**
+     * @param dataRepo   a repository storing the data of a GTFS dataset
+     * @param resultRepo a repository storing information about the validation process
+     * @param logger     a logger to log information about the validation process
+     */
+    public ValidateAgencyLangAndFeedInfoFeedLangMatch(final GtfsDataRepository dataRepo,
+                                                      final ValidationResultRepository resultRepo,
+                                                      final Logger logger) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+        this.logger = logger;
+    }
+
+    /**
+     * Use case execution method: checks if for each {@code Agency} defines a value for `agency.agency_lang` that
+     * matches the value contained in `feed_info.feed_lang` if `feed_info.txt` is provided. If this requirement is not
+     * met, a {@code FeedInfoLangAgencyLangMismatchNotice} is added to the {@code ValidationResultRepo} provided
+     * in the constructor.
+     */
+    public void execute() {
+        logger.info("Validating rule 'E055 - Mismatching feed and agency language fields'");
+
+        if (dataRepo.getFeedInfoAll().size() == 0) {
+            return;
+        }
+        // .get can be used without isPresent check here since line 55 ensures the presence of at least 1 element
+        // in the map
+        final String feedInfoFeedLang =
+                dataRepo.getFeedInfoAll().values().stream().findFirst().get().getFeedLang();
+        final Set<String> agencyLangCollection = new HashSet<>();
+        dataRepo.getAgencyAll().forEach((agencyId, agency) -> agencyLangCollection.add(agency.getAgencyLang()));
+        if (feedInfoFeedLang.equals(MUL)) {
+            // If feed_lang is mul and there isn't more than one agency_lang, that's an error
+            if (agencyLangCollection.size() <= 1) {
+                resultRepo.addNotice(new FeedInfoLangAgencyLangMismatchNotice(agencyLangCollection));
+                return;
+            }
+            return;
+        }
+        // If there is more than one agency_lang and feed_lang isn't mul, that's an error
+        // check if feed_lang is `mul` here is redundant but is provided for future software maintenance
+        if (!feedInfoFeedLang.equals(MUL) && agencyLangCollection.size() > 1 && !agencyLangCollection.contains(feedInfoFeedLang)) {
+            resultRepo.addNotice(new FeedInfoLangAgencyLangMismatchNotice(agencyLangCollection, feedInfoFeedLang));
+            return;
+        }
+
+        dataRepo.getAgencyAll().forEach((agencyId, agency) -> {
+            // If feed_lang is not mul and differs from agency_lang, that's an error
```

Comment:
I'll remove the redundant checks, since the unit tests fails if the initial `if (feedInfoFeedLang.equals(MUL)) {` is removed (it has been manually tested).